crates/sui-core/src/consensus_adapter.rs

```diff
@@ -239,50 +239,30 @@ impl ConsensusAdapter {
         ourselves: &AuthorityName,
         tx_digest: &TransactionDigest,
     ) -> Duration {
-        let (nominator, denominator) =
-            Self::position_submit_certificate(committee, ourselves, tx_digest);
-        const MAX_DELAY_MS: u64 = 60_000;
-        // There is no overflow as long as total_stake < u64::MAX / MAX_DELAY_MS, which seem to be reasonable expectation
-        let duration_ms = MAX_DELAY_MS * nominator / denominator;
-        Duration::from_millis(duration_ms)
+        let position = Self::position_submit_certificate(committee, ourselves, tx_digest);
+        const MAX_DELAY: Duration = Duration::from_secs(70);
+        const DELAY_STEP: Duration = Duration::from_secs(7);
+        let duration = DELAY_STEP * (position as u32/*num validators is less then u32::MAX*/);
+        std::cmp::min(duration, MAX_DELAY)
     }
 
     /// Returns a position of the current validator in ordered list of validator to submit transaction
-    /// You can think of this value as a float in range [0, 1)
-    /// Validators closer to 0 are submitting transaction sooner, and validators closer to 1 wait longer
-    ///
-    /// In practice, instead of returning float we return integer fraction here in form of (nominator, denominator)
-    /// If returned tuple (A, B) the value A is in range [0, B)
     fn position_submit_certificate(
```

Comment:
It will be useful to point to the high level algorithm or describe it here e.g. the goal of the delays, how positions are based on the order from transaction digest etc.